clientapp/src/components/Map.jsx

```diff
@@ -1,74 +1,179 @@
 import React from 'react';
 import { BingMap } from '../utilities/mapUtilities';
-/*
+/*setState
 Map props:
 mapInfo:
+cards:
 */
 export class Map extends React.Component {
     constructor(props) {
         super(props);
         this.loadMapScenario = this.loadMapScenario.bind(this);
-        this.state = {mapInfo: this.props.mapInfo};        
+        this.state = {mapInfo: this.props.mapInfo, cardInfo: []};       
     }
 
     componentDidMount(){
         window.onload = this.loadMapScenario;
     }
 
+    componentDidUpdate(){
+      //this.setState({cardInfo:() => this.props.cardInfo});
+      this.state.cardInfo = this.props.cardInfo;  
```

Comment:
state is typically used when something internal to the component may cause it to change. In this case, we are changing based on input from parent component, so it seems like we can just read directly from props, without reading from state. So componentDidUpdate() could look like:

`this.loadMapScenario()` and then in loadMapScenario, read from this.props.cardInfo

Reply:
sure